### ISSUE TITLE
Panic in nightly-msvc (cddc4a62d 2017-12-26), "thread 'rustc' panicked at 'failed to create span for type arguments',"

### ISSUE THREAD

**Issue by Centril** [rust-lang/rust-clippy#2306]
This is potentially a dupe, but I thought it better to overreport than to underreport.

Used on the following repo: https://github.com/Centril/proptest-arbitrary/commit/831cb4993dcd644c1fbb614b0ab83813fe0662f4

```
λ rustup default nightly-msvc
info: using existing install for 'nightly-x86_64-pc-windows-msvc'
info: default toolchain set to 'nightly-x86_64-pc-windows-msvc'

  nightly-x86_64-pc-windows-msvc unchanged - rustc 1.24.0-nightly (cddc4a62d 2017-12-26)

λ set RUST_BACKTRACE=1

λ cargo clippy --verbose
lib: proptest-arbitrary
       Fresh libc v0.2.30
       Fresh regex-syntax v0.4.1
       Fresh quick-error v1.2.1
       Fresh unicode-xid v0.0.4
       Fresh version_check v0.1.3
       Fresh lazy_static v0.2.11
       Fresh bit-vec v0.4.4
       Fresh frunk_core v0.0.21
       Fresh quote v0.3.15
       Fresh rand v0.3.18
       Fresh synom v0.11.3
       Fresh bit-set v0.4.0
       Fresh syn v0.11.11
       Fresh proptest v0.3.2
       Fresh frunk_derives v0.0.22
       Fresh derive_more v0.7.0
   Compiling proptest-arbitrary v0.2.1 (file:///D:/programming/proptest/proptest-arbitrary)
     Running `C:\Users\twing\.cargo\bin\clippy-driver rustc --crate-name proptest_arbitrary src\lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --emit=metadata --cfg "feature=\"cargo-clippy\"" --cfg "feature=\"default\"" --cfg "feature=\"frunk\"" --cfg "feature=\"frunk_core\"" --cfg "feature=\"frunk_derives\"" -C metadata=e9047cfbd99059c5 -C extra-filename=-e9047cfbd99059c5 --out-dir D:\programming\proptest\proptest-arbitrary\target\debug\deps -C incremental=D:\programming\proptest\proptest-arbitrary\target\debug\incremental -L dependency=D:\programming\proptest\proptest-arbitrary\target\debug\deps --extern proptest=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libproptest-a303a4f21c6b9b42.rlib --extern frunk_derives=D:\programming\proptest\proptest-arbitrary\target\debug\deps\frunk_derives-d5b8626e7534f410.dll --extern bit_set=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libbit_set-12a4d0976b7ce70b.rlib --extern derive_more=D:\programming\proptest\proptest-arbitrary\target\debug\deps\derive_more-3f056819d97b80fa.dll --extern frunk_core=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libfrunk_core-f0d1bcf4a443f064.rlib --cfg MIN_VER_1_24_0`
thread 'rustc' panicked at 'failed to create span for type arguments', src\libcore\option.rs:891:4
stack backtrace:
   0: std::sync::mpsc::sync::Queue::dequeue
   1: std::panicking::Location::column
   2: std::panicking::Location::column
   3: std::panicking::rust_panic_with_hook
   4: std::panicking::begin_panic_fmt
   5: std::panicking::begin_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: core::option::expect_failed
   9: <unknown>
  10: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_item
  11: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_mod
  12: rustc::hir::intravisit::NestedVisitorMap::inter
  13: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_item
  14: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_mod
  15: rustc::hir::intravisit::NestedVisitorMap::inter
  16: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_item
  17: <rustc::lint::context::LateContext<'a, 'tcx> as rustc::hir::intravisit::Visitor<'tcx>>::visit_mod
  18: rustc::hir::intravisit::NestedVisitorMap::inter
  19: rustc::lint::context::check_crate
  20: <env_logger::filter::Filter as core::fmt::Display>::fmt
  21: <env_logger::filter::Filter as core::fmt::Display>::fmt
  22: <rustc_driver::Compilation as core::fmt::Debug>::fmt
  23: rustc_driver::driver::compile_input
  24: rustc_driver::run_compiler
  25: <unknown>
  26: <unknown>
  27: _rust_maybe_catch_panic
  28: <unknown>
  29: std::sync::mpsc::sync::Queue::dequeue
  30: std::sys::windows::thread::Thread::new
  31: BaseThreadInitThunk
thread 'main' panicked at 'rustc_thread failed: Any', src\libcore\result.rs:916:4
stack backtrace:
   0: std::sync::mpsc::sync::Queue::dequeue
   1: std::panicking::Location::column
   2: std::panicking::Location::column
   3: std::panicking::rust_panic_with_hook
   4: std::panicking::begin_panic_fmt
   5: std::panicking::begin_panic_fmt
   6: rust_begin_unwind
   7: core::panicking::panic_fmt
   8: <unknown>
   9: <unknown>
  10: _rust_maybe_catch_panic
  11: std::rt::lang_start
  12: <unknown>
  13: BaseThreadInitThunk
error: Could not compile `proptest-arbitrary`.

Caused by:
  process didn't exit successfully: `C:\Users\twing\.cargo\bin\clippy-driver rustc --crate-name proptest_arbitrary src\lib.rs --crate-type lib --emit=dep-info,link -C debuginfo=2 --emit=metadata --cfg feature="cargo-clippy" --cfg feature="default" --cfg feature="frunk" --cfg feature="frunk_core" --cfg feature="frunk_derives" -C metadata=e9047cfbd99059c5 -C extra-filename=-e9047cfbd99059c5 --out-dir D:\programming\proptest\proptest-arbitrary\target\debug\deps -C incremental=D:\programming\proptest\proptest-arbitrary\target\debug\incremental -L dependency=D:\programming\proptest\proptest-arbitrary\target\debug\deps --extern proptest=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libproptest-a303a4f21c6b9b42.rlib --extern frunk_derives=D:\programming\proptest\proptest-arbitrary\target\debug\deps\frunk_derives-d5b8626e7534f410.dll --extern bit_set=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libbit_set-12a4d0976b7ce70b.rlib --extern derive_more=D:\programming\proptest\proptest-arbitrary\target\debug\deps\derive_more-3f056819d97b80fa.dll --extern frunk_core=D:\programming\proptest\proptest-arbitrary\target\debug\deps\libfrunk_core-f0d1bcf4a443f064.rlib --cfg MIN_VER_1_24_0` (exit code: 101)
```

**Comment by oli-obk** [comment]
This backtrace looks weird. It doesn't seem to be caused by clippy code, but inside rustc. 😕

**Comment by mati865** [comment]
Commit `831cb49` doesn't crash with nightly on x86_64 Linux, either it was fixed at some point or it's Rust bug.

**Comment by Areredify** [comment]
It no longer crashes and presumably was fixed by pr #3151, can we close this @flip1995 ?
